kernel: bump 6.6 to 6.6.77

graysky2 opened the pull request:
kernel: bump 6.6 to 6.6.77

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.77

No patches automatically rebased/checksum update only

Build system: x86/64
Build-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

graysky2 commented:
I do not have time to debug right now. If someone else can take a look at why this failed to boot - at least for bcm2712 - that would be great. Use this PR with caution.

mgz0227 commented:
I have it running fine on x86 with no issues so far

namiltd commented:
Completely erase the SD card and re-flash.

graysky2 commented:
> Completely erase the SD card and re-flash. 
I did, no change

zxlhhyccc commented:
````
=== Checking commit '86b902e294ecf781ff77a27290aa90436bea0171'
Author name (John Audia) seems ok
Commit subject line seems ok (kernel: bump 6.6 to 6.6.75)
Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: John Audia <therealgraysky@proton.me>')
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.75

All patches automatically rebased.

Build system: x86/64
Build-tested: bcm27xx/bcm2712
Run-tested: Not run-tested due to boot breakage[1]

1. https://forum.openwrt.org/t/recent-build-of-openwrt-breaks-booting-on-rpi5b/223449
A commit message exists
Error: Process completed with exit code 1.
`````

PerkelSimon commented:
no issue on flint2 mt-6000

namiltd commented:
Draft version 6.6.76 can be found here: https://github.com/openwrt/openwrt/pull/16547/commits/d4dac695bf79d9bae60c64c41b00f5b21abb765b
I only did refresh for a few platforms that are essential to me, so please finish it.

graysky2 commented:
@namiltd - I refreshed earlier this AM but needed to test before pushing.

namiltd commented:
Since there are compilation errors for quantummax after https://github.com/openwrt/openwrt/commit/4b6886d9fd53c3d15fcfda4c36cbfa8d727bed55 I think you should temporarily add commits from https://github.com/openwrt/openwrt/pull/17908

graysky2 commented:
You're recommending that I pull into this PR the two commits from https://github.com/openwrt/openwrt/pull/17908?

namiltd commented:
I don't know if both are necessary but once they are approved you will remove them.

hauke commented on target/linux/imx/patches-6.6/600-PCI-imx6-Start-link-at-max-gen-first-for-IMX8MM-and-IMX8MP.patch:
```diff
@@ -100,20 +100,22 @@ Signed-off-by: Tim Harvey <tharvey@gateworks.com>
 -		.flags = IMX6_PCIE_FLAG_SUPPORTS_SUSPEND,
 +		.flags = IMX6_PCIE_FLAG_SUPPORTS_SUSPEND |
-+			 IMX6_PCIE_FLAG_GEN1_LAST,
++			IMX6_PCIE_FLAG_GEN1_LAST,
```
Why do you change this?

graysky2 replied:
I removed one of the tabs

hauke commented:
For the review it would be much easier if you would refresh the qualcommbe target with the old kernel before.

graysky2 commented on target/linux/imx/patches-6.6/600-PCI-imx6-Start-link-at-max-gen-first-for-IMX8MM-and-IMX8MP.patch:
```diff
@@ -100,20 +100,22 @@ Signed-off-by: Tim Harvey <tharvey@gateworks.com>
 -		.flags = IMX6_PCIE_FLAG_SUPPORTS_SUSPEND,
 +		.flags = IMX6_PCIE_FLAG_SUPPORTS_SUSPEND |
-+			 IMX6_PCIE_FLAG_GEN1_LAST,
++			IMX6_PCIE_FLAG_GEN1_LAST,
```
I removed one of the tabs

graysky2 commented:
> For the review it would be much easier if you would refresh the qualcommbe target with the old kernel before.

Are you asking me to redo the refresh for 6.6.75 against HEAD?

Ansuel commented:
@hauke ok i will check it

graysky2 committed:
kernel: bump 6.6 to 6.6.76

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.76

Manually rebased:
	bcm27xx/patches-6.6/950-0519-usb-dwc3-Set-DMA-and-coherent-masks-early.patch
	imx/patches-6.6/600-PCI-imx6-Start-link-at-max-gen-first-for-IMX8MM-and-IMX8MP.patch

Removed upstreamed:
	bcm27xx/patches-6.6/950-1446-media-i2c-ov9282-Correct-the-exposure-offset.patch[1]
	bcm47xx/patches-6.6/701-bgmac-reduce-max-frame-size-to-support-just-MTU-1500.patch[2]
	bcm53xx/patches-6.6/700-bgmac-reduce-max-frame-size-to-support-just-MTU-1500.patch[3]
	ramips/patches-6.6/003-v6.14-clk-ralink-mtmips-remove-duplicated-xtal-clock-for-Ralink.patch[4]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.76&id=11c7649c9ec3dcaf0a7760551ad30747d9e02d81
2, 3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.76&id=5e6e723675e54ced5200bcc367e2526badc4070c
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.76&id=d0edcd0d18d700d76c61c091a24568b8b8c3b387

Build system: x86/64
Build-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17822
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented:
Create an additional commit with the result of "make target/linux/refresh"
and only then "bump 6.6 to 6.6.75" and bump 6.6 to 6.6.76".

graysky2 commented:
@namiltd - running `make target/linux/refresh` on HEAD made no changes/nothing to commit

hauke commented:
> @namiltd - running `make target/linux/refresh` on HEAD made no changes/nothing to commit

It should result in changes to the patches in the `qualcommbe` target.

graysky2 committed:
qualcommb/ipq95xx: refresh patches ahead of 6.6.75

Refreshed patches for qualcommb/ipq95xx by running
make target/linux/refresh after creating a .config containing:
CONFIG_TARGET_qualcommbe=y
CONFIG_TARGET_qualcommbe_ipq95xx=y
CONFIG_TARGET_qualcommbe_ipq95xx_DEVICE_qcom_rdp433=y

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17822
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
I see now... I had to create a .config containing the following then run that command:
```
    CONFIG_TARGET_qualcommbe=y
    CONFIG_TARGET_qualcommbe_ipq95xx=y
    CONFIG_TARGET_qualcommbe_ipq95xx_DEVICE_qcom_rdp433=y
```

graysky2 committed:
kernel: bump 6.6 to 6.6.75

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.75

All patches automatically rebased.

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17822
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

Ansuel commented:
@hauke if you are ok with it I would merge https://github.com/openwrt/openwrt/pull/17788 and include in that pr the kernel patch refresh. What do you think?

Ansuel commented:
ok also that works :D

hauke commented:
@Ansuel If you think #17788 is ready you can merge it and include a refresh. We can then rebase this PR on top of it.

AgustinLorenzo commented:
Hi @graysky2,

I am seeing this in qualcommax:

`make[5]: Entering directory '/home/runner/actions-runner/_work/openwrt/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/linux-6.6.76'
  CC      net/ipv4/ipmr.o
net/ipv4/ipmr.c: In function 'ipmr_mfc_stats_update':
net/ipv4/ipmr.c:455:35: error: invalid operands to binary + (have 'atomic_long_t' {aka 'atomic64_t'} and 'u64' {aka 'long long unsigned int'})
  455 |         cache->_c.mfc_un.res.pkt  += pkts_out;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~  ^~
      |                             |
      |                             atomic_long_t {aka atomic64_t}
net/ipv4/ipmr.c:456:36: error: invalid operands to binary + (have 'atomic_long_t' {aka 'atomic64_t'} and 'u64' {aka 'long long unsigned int'})
  456 |         cache->_c.mfc_un.res.bytes += bytes_out;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~
      |                             |
      |                             atomic_long_t {aka atomic64_t}
make[9]: *** [scripts/Makefile.build:243: net/ipv4/ipmr.o] Error 1
make[8]: *** [scripts/Makefile.build:480: net/ipv4] Error 2
make[7]: *** [scripts/Makefile.build:480: net] Error 2`

Can you check?

Regards, Agustin

namiltd commented:
@AgustinLorenzo attach your .config

graysky2 commented:
@AgustinLorenzo - I am afraid that is something @Ansuel should check out. I just refreshed the patches for that targets based on his commits which were recently merged.

AgustinLorenzo commented:
> @AgustinLorenzo attach your .config

Hi @namiltd,

Here you my .full_config: https://github.com/AgustinLorenzo/openwrt/blob/main_nss/.full_config

I am backing to 6.6.74 for check if compile fine.

Link: https://github.com/AgustinLorenzo/openwrt/actions/runs/13229435510/job/36924507938

Regards, Agustín

namiltd commented:
Could you please do a test build for 6.6.75?

AgustinLorenzo commented:
> @AgustinLorenzo Could you please do a test build for 6.6.75?

Okay, preparing again but with 6.6.75

Thanks

namiltd commented:
You can cancel the 6.6.75 kernel test. I see that in kernel 6.6.76 there were changes in the ipmr.c file.

AgustinLorenzo commented:
> You can cancel the 6.6.75 kernel test. I see that in kernel 6.6.76 there were changes in the ipmr.c file.

Okay, canceled, if you need something, tell me :)

AgustinLorenzo commented:
> Unfortunately, in the configuration there is CONFIG_TARGET_MULTI_PROFILE=y I need to find which router this error is for. There are some additional patches being uploaded for it that are incompatible with 6.6.76. I suspect the MAC80211_MESH package.

Hi @namiltd,

I am checking, if is MAC802211_MESH is from NSS and is outside of OpenWRT.

Regards, Agustin

namiltd commented:
So the PR is ready for approval.

AgustinLorenzo commented:
Yeah is ready for push :)

graysky2 committed:
kernel: bump 6.6 to 6.6.77

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.77

No patches automatically rebased/checksum update only

Build system: x86/64
Build-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: bcm27xx/bcm2712, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17822
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rmandrad commented:
hopefully the .77 is a quick one to turnaround as it is just a revert

King-Of-Knights commented:
@AgustinLorenzo please check this [patch](https://github.com/qosmio/openwrt-ipq/commit/7c4dcd77b2d03a11f8a6c750ba4d749fecc055c6#diff-362084bcfb982d917f2b700f6f033cdd122dc79ef4a3ea4c4c7e916c8e44721f)
